xdsl/dialects/memref.py

```diff
@@ -271,5 +276,33 @@ def get(sym_name: str | StringAttr,
             })
 
 
-MemRef = Dialect([Load, Store, Alloc, Alloca, Dealloc, GetGlobal, Global],
+@irdl_op_definition
+class Dim(Operation):
+    name = "memref.dim"
+
+    source: Annotated[Operand, MemRefType]
```

Comment:
Shouldn't this be `MemRefType[Attribute]` instead?
`MemRefType` is a generic type.